util/conditions/compare.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+// Check compares the actual and expected values provided using the
```

Comment:
I'm curious what's the motivation for this? are we unexpectedly panicking somewhere atm?

Reply:
@enxebre I added this to catch panic cases if any occurs 

it was mentioned in the issue discussion

https://github.com/kubernetes-sigs/cluster-api/issues/8186#issuecomment-1467962898